docs/tutorials/openshift.md

```diff
@@ -2,6 +2,60 @@
 This tutorial describes how to configure ExternalDNS to use the OpenShift Route source.
 It is meant to supplement the other provider-specific setup tutorials.
 
+### For OCP 4.x
+
+In OCP 4.x, if you have multiple ingress controllers then you must specify an ingress controller name or a router name(you can get it from the route's Status.Ingress.RouterName field).
+If you don't specify an ingress controller's or router name when you have multiple ingresscontrollers in your environment then the route gets populated with multiple entries of router canonical hostnames which causes external dns to create a CNAME record with multiple router canonical hostnames pointing to the route host which is a violation of RFC 1912 and is not allowed by Cloud Providers which leads to failure of record creation.
+Once you specify the ingresscontroller or router name then that will be matched by the external-dns and the router canonical hostname corresponding to this routerName(which is present in route's Status.Ingress.RouterName field) is selected and a CNAME record of this route host pointing to this router canonical hostname is created.
+
+Your externaldns CR shall be created as per the following example.
+Replace names in the domain section and zone ID as per your environment.
+This is example is for AWS environment.
+
+```yaml
+
+  apiVersion: externaldns.olm.openshift.io/v1alpha1
+  kind: ExternalDNS
+  metadata:
+    name: sample1
+  spec:
+    domains:
+      - filterType: Include
+        matchType: Exact
+        names: apps.miheer.externaldns
+    provider:
+      type: AWS
+    source:
+      hostnameAnnotation: Allow
+      openshiftRouteOptions:
+        routerName: default
+      type: OpenShiftRoute
+    zones:
+      - Z05387772BD5723IZFRX3
+
+```
```

Comment:
This custom resource is specific to ExternalDNS Operator, it doesn't make much sense in this (upstream) repository. It can only raise the questions which this repository cannot answer.

Reply:
I will address this in the next PR. In OCP4 we support external dns via operator only. You are concerned about OKD users which is fine so I will address that in another PR.